functions/internal/util/util.go

```diff
@@ -176,3 +181,53 @@ func ReadCryptoRandBytes(b []byte) {
 		panic(fmt.Errorf("could not read random bytes: %v", err))
 	}
 }
+
+// CheckHTTPS retrieves the scheme from the X-Forwarded-Proto or RFC7239
+/*
+We do this because in the function running in a cloud container the SSL termination
+has likely happened upstream so we need to check the headers to reject HTTP only.
```

Comment:
I would replace "has likely" with "has"; this codebase doesn't perform TLS termination, so there's no way that it hasn't happened upstream.